diff --git a/app/build.gradle b/app/build.gradle
index 3f9ad1d..c815981 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -2,6 +2,7 @@ apply plugin: 'com.android.application'
 apply plugin: 'com.android.databinding'
 apply plugin: 'com.neenbedankt.android-apt'
 apply plugin: 'ext-properties'
+apply plugin: 'me.tatarka.retrolambda'
 
 android {
     compileSdkVersion rootProject.ext.compileSdkVersion
@@ -17,6 +18,11 @@ android {
         testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
     }
 
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+
     signingConfigs {
         // debug {
         //     storeFile rootProject.file("debug.keystore")
@@ -52,6 +58,15 @@ android {
     }
 }
 
+retrolambda {
+    jdk System.getenv("JAVA8_HOME")
+    oldJdk System.getenv("JAVA7_HOME")
+    javaVersion JavaVersion.VERSION_1_7
+    //jvmArgs '-arg1', '-arg2'
+    defaultMethods false
+    incremental true
+}
+
 repositories {
     // for stetho-realm
     maven {
diff --git a/app/proguard-rules.pro b/app/proguard-rules.pro
index 4412c74..1765490 100644
--- a/app/proguard-rules.pro
+++ b/app/proguard-rules.pro
@@ -24,3 +24,6 @@
 
 # for Stetho
 -keep class com.facebook.stetho.**{ *; }
+
+# for retrolambda
+-dontwarn java.lang.invoke.*
\ No newline at end of file
diff --git a/app/src/androidTest/java/com/example/realm_template/MainActivityTest.java b/app/src/androidTest/java/com/example/realm_template/MainActivityTest.java
index bf3bcf2..e3c5431 100644
--- a/app/src/androidTest/java/com/example/realm_template/MainActivityTest.java
+++ b/app/src/androidTest/java/com/example/realm_template/MainActivityTest.java
@@ -54,12 +54,9 @@ public class MainActivityTest {
         final Realm realm = Realm.getInstance(configuration);
         //noinspection TryFinallyCanBeTryWithResources
         try {
-            realm.executeTransaction(new Realm.Transaction() {
-                @Override
-                public void execute(Realm realm) {
-                    insertInTransaction(realm, "foo", 25);
-                    insertInTransaction(realm, "bar", 0);
-                }
+            realm.executeTransaction(r -> {
+                insertInTransaction(r, "foo", 25);
+                insertInTransaction(r, "bar", 0);
             });
         } finally {
             realm.close();
diff --git a/app/src/main/java/com/example/realm_template/MainActivity.java b/app/src/main/java/com/example/realm_template/MainActivity.java
index 77eff08..616c023 100644
--- a/app/src/main/java/com/example/realm_template/MainActivity.java
+++ b/app/src/main/java/com/example/realm_template/MainActivity.java
@@ -168,27 +168,21 @@ public class MainActivity extends AppCompatActivity {
         user2.setName("Jiro Yamada");
         user2.setAge(20 + random.nextInt(20));
 
-        realm.executeTransaction(new Realm.Transaction() {
-            @Override
-            public void execute(Realm realm) {
-                realm.allObjects(User.class).clear();
+        realm.executeTransaction(realm1 -> {
+            realm1.allObjects(User.class).clear();
 
-                realm.copyToRealm(user1);
-                realm.copyToRealm(user2);
-            }
+            realm1.copyToRealm(user1);
+            realm1.copyToRealm(user2);
         });
     }
 
     private void incrementAge() {
         final RealmResults<User> allUsers = realm.allObjects(User.class);
-        realm.executeTransaction(new Realm.Transaction() {
-            @Override
-            public void execute(Realm realm) {
-                // don't use for-each. see https://github.com/realm/realm-java/issues/640
-                for (int i = 0; i < allUsers.size(); i++) {
-                    final User user = allUsers.get(i);
-                    user.setAge(user.getAge() + 1);
-                }
+        realm.executeTransaction(r -> {
+            // don't use for-each. see https://github.com/realm/realm-java/issues/640
+            for (int i = 0; i < allUsers.size(); i++) {
+                final User user = allUsers.get(i);
+                user.setAge(user.getAge() + 1);
             }
         });
     }
diff --git a/build.gradle b/build.gradle
index d56409b..e11dda1 100644
--- a/build.gradle
+++ b/build.gradle
@@ -8,10 +8,15 @@ buildscript {
         classpath 'com.android.tools.build:gradle:1.3.1'
         classpath 'com.android.databinding:dataBinder:1.0-rc2'
         classpath 'com.neenbedankt.gradle.plugins:android-apt:1.6'
+        classpath 'me.tatarka:gradle-retrolambda:3.2.3'
+        classpath 'me.tatarka.retrolambda.projectlombok:lombok.ast:0.2.3.a2'
 
         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
     }
+
+    // Exclude the version that the android plugin depends on.
+    configurations.classpath.exclude group: 'com.android.tools.external.lombok'
 }
 
 allprojects {
